experimental/Origami/src/Origami.jl

```diff
@@ -0,0 +1,333 @@
+ordered) non-zero degrees of the singularities. For a singularity of cone angle 2kπ the
+degree of the singularity is k-1.
+
+# Examples
+```jldoctest
+julia> o = origami(cperm([1,2],[3,4]), cperm([2,5,3,6],[4,7]))
+Origami ((1,2)(3,4),(2,5,3,6)(4,7), 7)
+
+julia> stratum(o)
+3-element Vector{Int64}:
+ 2
+ 1
+ 1
+```
+"""
+function stratum(o::Origami)
+    h = horizontal_perm(o)
+    v = vertical_perm(o)
+    commutator = comm(h, v)
+    cycs = cycles(commutator)
+    unsorted_stratum = [length(c) - 1 for c in cycs if length(c) > 1]
+    return sort(unsorted_stratum, rev=true)
+end
+
+@doc raw"""
+    genus(o::Origami)
+
+Calculates the genus of an origami.
+
+# Examples
+```jldoctest
+julia> o = origami(cperm([1,2]), cperm([2,3]))
+Origami ((1,2),(2,3), 3)
+
+julia> genus(o)
+2
+```
+"""
+function genus(o::Origami)
+    return ZZ((sum(stratum(o)) + 2) * 0.5)
+end
+
+function veech_group(O::Origami)
+    # TODO use hashtables or not? Implement modular subgroup?
+    GAP.Globals.ComputeVeechGroupWithHashTables(GapObj(O))
+end
+
+function index_monodromy_group(o::Origami)
+    return GAP.Globals.IndexOfMonodromyGroup(GapObj(o))::Int
+end
+
+function sum_of_lyapunov_exponents(o::Origami)
+    return QQFieldElem(GAP.Globals.SumOfLyapunovExponents(GapObj(o)))
+end
+
+function translations(o::Origami)
+    h = horizontal_perm(o)
+    v = vertical_perm(o)
+    G = normalform_conjugators(o)
+
+    O = (i -> origami(i^-1 * h * i, i^-1 * v * i)).(G)
+    list = [cperm([degree(o)])]
+
+    l = length(O)
+    for i in 1:l
+        positions = findall(item -> item == O[i], O)
+        if length(positions) != 1
+            for j in positions
+                push!(list, G[i] * G[j]^-1)
+            end
+        end
+    end
+
+    return collect(Set(list))
+end
+
+function is_hyperelliptic(o::Origami)
+    # check whether -1 is in the veech group
+	if !veech_group_is_even(o) 
+        return false
+    end
+
+    n = 2
+    n_inv = 1 / 2
+	x = horizontal_perm(o)
+	y = vertical_perm(o)
+	g = genus(o)
+
+    L = point_reflections(o)
+	L = filter(i -> order(i) == 2, L)
+
+	if L == []
+        return false
+    end
+
+    # helper to find all elements in x that are not in y
+    function list_diff(x, y)
+        set_1 = Set(x)
+        set_2 = Set(y)
+        diff = setdiff(set_1, set_2)
+        return collect(diff)
+    end
+
+    degree_list = collect(1:degree(o))
+    for sigma in L
+        # fixpoints
+        b = 0
+        b = b + length(list_diff(degree_list, moved_points(sigma)))
+        b = b + length(list_diff(degree_list, moved_points(sigma*x)))
+        b = b + length(list_diff(degree_list, moved_points(sigma*y)))
+
+        for i in degree_list
+            if i^(sigma * x^-1 * y^-1) == i^(y*x*(x*y)^-1)
+                b = b + 1
+            end
+        end
+
+        # TODO can n_inv yields floating point errors?
+        if n_inv * (g - 1 - b * 0.5) + 1 == 0
```

Comment:
You set `n_inv` to `1/2` which is just `0.5` so yeah there can be floating point errors. I am not sure what this code is *meant* to be doing, but we can try to refactor it to work without floating point?

First multiply by $n=2$:
```suggestion
        if (g - 1 - b * 0.5) + 2 == 0
```
then simplify
```suggestion
        if g + 1 - b * 0.5 == 0
```
and multiply by 2 again
```suggestion
        if 2*g + 2 - b == 0
```
or
```suggestion
        if 2*g + 2 == b
```

Now is that what you want?